django/contrib/postgres/aggregates/general.py

```diff
@@ -5,7 +5,43 @@
 ]
 
 
-class ArrayAgg(Aggregate):
+class OrderableAgg(Aggregate):
+    template = "%(function)s(%(expressions)s %(ordering)s)"
+
+    def __init__(self, expression, ordering=None, **extra):
+        super(OrderableAgg, self).__init__(
+            expression,
+            ordering=ordering,
+            **extra
+        )
+
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False,
+                           for_save=False):
+        # resolve the ordering if it's there
+        ordering = self.extra.pop('ordering', None)
+        if ordering is not None:
+            parsed_ordering = self._parse_expressions(ordering)[0]
+            resolved_ordering = parsed_ordering.resolve_expression(query, allow_joins, reuse,
+                                                                   summarize)
+
+            self.extra['ordering'] = resolved_ordering
+
+        return super(OrderableAgg, self).resolve_expression(query, allow_joins, reuse, summarize,
+                                                            for_save)
```

Comment:
I think storing ordering in extra is ok unless it needs to be retrieved in `get_source_expressions` for some reason. I don't think it does though.

As long as the ordering parameter is exposed on `__init__` I think that's fine.